draft-ietf-dnsop-svcb-https.md

```diff
@@ -784,14 +784,14 @@ default transports.
 Clients MUST include an `application_layer_protocol_negotiation` extension
 in their ClientHello with a ProtocolNameList that includes at least one ID
 from the ALPN set.  Clients SHOULD also include any other values that they
-support and could negotiate on that connection with equivalent or better
-security properties.  For example, if the ALPN set only contains "http/1.1",
+are willing to negotiate on that connection.
+For example, if the ALPN set only contains "http/1.1",
 the client could include "http/1.1" and "h2" in the ProtocolNameList.
 
 Once the client has formulated the ClientHello, protocol negotiation
 on that connection proceeds as specified in {{!ALPN}}, without regard to the
 SVCB ALPN set.  To preserve the security guarantees of this process, clients
-MUST consolidate all compatible ALPN IDs into a single ProtocolNameList.
+MUST consolidate all compatible ALPN IDs into a single ClientHello.
```

Comment:
I'm not quite following this sentence (old and new version). Is it saying the same thing as the SHOULD above or something else?

Reply:
It's kind of the inverse.  There are really two requirements:
* If you use an ALPN from the list, you SHOULD accompany it with any other ALPN you would like to do on that transport, whether or not it's in the list.
* You MUST only form one ClientHello per transport in total, so if you like two ALPNs from the list, they have to go in the same ClientHello (if that is possible).

The first requirement defends against attackers on the DNS path, and the second defends against attackers on the network path.

That said, these requirements are effectively kind of redundant, so we could possibly simplify this text.  Suggestions welcome.

Reply:
Ah. I guess my interpretation was that, if `ClientHelloFromSVCB(proto=h2)` and `ClientHelloFromSVCB(proto=http/1.1)` both result in the same ClientHello (based on the rule that you include all of them), that's fine whether I connect to them serially or in parallel or whatever.

What does consolidating mean if, say, I see SVCB records for h2 and http/1.1 but other properties, like ECH keys or addresses, are different? I figured the model would be that you pick which one(s) you want, and send out ClientHellos for them. If you decided you wanted to race the http/1.1 and h2 records, they'll end up sending the same ALPN. If you decided you wanted to race an h3 one in there, it will be different. When picking them, you probably do need to be aware of the correspondence, but I think the impact is less security and more making your connection strategy coherent.

Reply:
OK, evidently the previous text was not clear enough.  I've reorganized it; hopefully its less ambiguous now.